source/Esf.Website/App/common/interfaces/esfStateProvider.interface.ts

```diff
@@ -0,0 +1,5 @@
+﻿import { EsfStateViewModel } from '../models/esfStateViewModel';
+
+export interface EsfStateProvider {
+    getState(): EsfStateViewModel;
+}
```

Comment:
Why do we need this EsfStateProvider instead of passing directly the state where it is needed? From the looks of it services are indirectly using the EsFiddlerComponent's state. I think only services should be talking in such a way between themselves. If we need such state service we could probably think about Observable of state and listeners attaching to it, but do we need it?